components/email/InteractiveSection.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+import { useContextTheme } from '../ThemeContext';
+
+export default function InteractiveSection({ children }) {
+  const [componentPointer, setComponentPointer] = useState(0);
+  const [data, setData] = useState({});
+  const { cardsBackgroundColor } = useContextTheme();
+
+  const updateData = (childData) => {
+    setData((prevState) => {
+      return { ...prevState, ...childData };
+    });
+  };
+
+  const nextComponent = () => {
+    setComponentPointer(componentPointer + 1);
```

Comment:
```suggestion
    setComponentPointer((componentPointer) => componentPointer + 1);
```

Let's do this to protect against stale data.